src/cluster_legacy.h

```diff
@@ -344,6 +344,29 @@ static_assert(offsetof(clusterMsgLight, data) == 16, "unexpected field offset");
 
 #define CLUSTERMSG_LIGHT_MIN_LEN (sizeof(clusterMsgLight) - sizeof(union clusterMsgData))
 
+typedef struct {
+    char sig[4];       /* Signature "RCmb" (Cluster message bus). */
+    uint32_t totlen;   /* Total length of this message */
+    uint16_t ver;      /* Protocol version, currently set to CLUSTER_PROTO_VER. */
+    uint16_t notused1; /* full: port, light: notused1 */
+    uint16_t type;     /* Message type */
+    uint16_t notused2; /* full: count, light: notused2 */
+} clusterMsgUnknown;
```

Comment:
```suggestion
} clusterMsgHeader;
```
Not a very strong preference, but unknown isn't true, because at this point we do know what type it is (type is filled here) we just haven't parsed it yet to create the right structure. You can tell this was sort of always in the intention since hdr (stands for header) is used in most of the places you are creating this struct.

As a side effect, all the hdr -> hdr_unknown also all go away.